tools/summaries/palm.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+import re
+
+import dc
+import requests
+
+_API_KEY = os.getenv("PALM_API_KEY")
+_API_URL = "https://generativelanguage.googleapis.com/v1beta2/models/text-bison-001:generateText"
+_TEMPERATURE = 0.2
+# Sometimes the summaries may not be factual. Retry them in that case.
+# We use a heuristic that the summary must contain at least 2 numbers to be factual.
```

Comment:
we could also verify that the returned numbers matches a value in the prompt csv.